sdk/keyvault/azure-security-keyvault-keys/src/cryptography/cryptography_client.cpp

```diff
@@ -56,340 +60,166 @@ inline std::vector<uint8_t> CreateDigest(
 
 CryptographyClient::~CryptographyClient() = default;
 
-void CryptographyClient::Initialize(std::string const&, Azure::Core::Context const& context)
-{
-  if (m_provider != nullptr)
-  {
-    return;
-  }
-
-  try
-  {
-    auto key = m_remoteProvider->GetKey(context).Value;
-    m_provider = LocalCryptographyProviderFactory::Create(key);
-    if (m_provider == nullptr)
-    {
-      // KeyVaultKeyType is not supported locally. Use remote client.
-      m_provider = m_remoteProvider;
-      return;
-    }
-  }
-  catch (Azure::Core::RequestFailedException const& e)
-  {
-    if (e.StatusCode == HttpStatusCode::Forbidden)
-    {
-      m_provider = m_remoteProvider;
-    }
-    else
-    {
-      throw;
-    }
-  }
-}
-
 CryptographyClient::CryptographyClient(
     std::string const& keyId,
     std::shared_ptr<Core::Credentials::TokenCredential const> credential,
-    CryptographyClientOptions const& options,
-    bool forceRemote)
+    CryptographyClientOptions const& options)
 {
   auto apiVersion = options.Version.ToString();
-  m_keyId = keyId;
-  m_remoteProvider = std::make_shared<RemoteCryptographyClient>(keyId, credential, options);
-  m_pipeline = m_remoteProvider->Pipeline;
-
-  if (forceRemote)
+  m_keyId = Azure::Core::Url(keyId);
```

Comment:
If it is a good idea to change this type from string to url, maybe rename the field to something like `m_keyIdUrl`.
```suggestion
  m_keyIdUrl = Azure::Core::Url(keyId);
```